common/flatpak-utils.c

```diff
@@ -1140,6 +1140,213 @@ flatpak_compare_ref (const char *ref1, const char *ref2)
   return 0;
 }
 
+static char *
+line_get_word (char **line)
+{
+  char *word = NULL;
+
+  while (g_ascii_isspace (**line))
+    (*line)++;
+
+  if (**line == 0)
+    return NULL;
+
+  word = *line;
+
+  while (**line && !g_ascii_isspace (**line))
+    (*line)++;
+
+  if (**line)
+    {
+      **line = 0;
+      (*line)++;
+    }
+
+  return word;
+}
+
+static char *
+glob_to_regexp (const char *glob, GError **error)
+{
+  g_autoptr(GString) regexp = g_string_new ("");
+  int parts = 1;
+
+  if (g_str_has_prefix (glob, "app/"))
+    {
+      glob += strlen ("app/");
+      g_string_append (regexp, "app/");
+    }
+  else if (g_str_has_prefix (glob, "runtime/"))
+    {
+      glob += strlen ("runtime/");
+      g_string_append (regexp, "runtime/");
+    }
+  else
+    g_string_append (regexp, "(app|runtime)/");
+
+  /* We really need an id part, the rest is optional */
+  if (*glob == 0)
+    {
+      flatpak_fail_error (error, FLATPAK_ERROR_INVALID_DATA, _("Empty glob"));
```

Comment:
Would be helpful to show the erroneous input here (ideally, a line number too).

Reply:
The line number is added in the caller.